src/validate.ts

```diff
@@ -50,6 +51,26 @@ export function validateTransfer(
   });
 }
 
+export function validateRemittance( remittanceRequest: RemittanceRequest
+): Promise<void> {
+  const { amount, currency, payee, referenceId }: RemittanceRequest = remittanceRequest || {};
+  return Promise.resolve().then(() => {
+    strictEqual(isTruthy(referenceId), true, "referenceId is required");
+    strictEqual(isUuid4(referenceId as string), true, "referenceId must be a valid uuid v4");
```

Comment:
why the need to coerce this?

Reply:
because we're using the uuid's as strings through the code base e.g take a look at src/remittances line 147

Reply:
isn't it already a string? do we need to coerce?

Reply:
a better explanation about the coercion in this case. 

in the `RemittanceRequest` interface, the `referenceId`  was optional hence it's type was either `string` or `undefined`. 
so the coercion was needed in order to test for equality on a value that's potentially undefined. 

i have changed the `referenceId`'s data type to strictly a string in the fix i'm sending in a few minutes. 

thanks!

Reply:
sent!